File: server/app/src/main/java/com/borkdominik/big/glsp/uml/core/pattern/PhasePatternAction.java
/********************************************************************************
 * Copyright (c) 2024 borkdominik and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
 * available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: EPL-2.0 OR MIT
 ********************************************************************************/
package com.borkdominik.big.glsp.uml.core.pattern;

import com.borkdominik.big.glsp.server.core.handler.action.new_file.BGNewFileResponseAction;
import org.eclipse.glsp.server.actions.RequestAction;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class PhasePatternAction extends RequestAction<BGNewFileResponseAction> {
   public static final String KIND = "phasePattern";

   private final Map<String, String> options;
   private final List<String> phases;

   public PhasePatternAction() {
      super(KIND);
      options = new HashMap<>();
      phases = new ArrayList<>();
   }

   public PhasePatternAction(final Map<String, String> options, List<String> phases) {
      super(KIND);
      this.options = options;
      this.phases = phases;
   }

   public Map<String, String> getOptions() { return options; }

   public List<String> getPhases() {
      return phases;
   }
}

File: client/packages/uml-protocol/src/action-protocol/phase-pattern.action.ts
/*********************************************************************************
 * Copyright (c) 2023 borkdominik and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the MIT License which is available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 *********************************************************************************/
import { Action, JsonMap, RequestAction, ResponseAction } from '@eclipse-glsp/protocol';

export interface PhasePatternAction extends RequestAction<PhasePatternResponseAction> {
    kind: typeof PhasePatternAction.KIND;
    options?: JsonMap;
    phases: string[];
}
export namespace PhasePatternAction {
    export const KIND = 'phasePattern';

    export function create(idPorvider: string, phases: string[]): PhasePatternAction {
        return {
            kind: KIND,
            options: {
                idPorvider: idPorvider
            },
            phases: phases,
            requestId: ''
        };
    }
}

export interface PhasePatternResponseAction extends ResponseAction {
    kind: typeof PhasePatternResponseAction.KIND;
    sourceUri: string;
}

export namespace PhasePatternResponseAction {
    export const KIND = 'phasePatternResponse';

    export function is(action: unknown): action is PhasePatternResponseAction {
        return Action.hasKind(action, KIND);
    }
}
